app/assets/javascripts/controllers/cloud_subnet/cloud_subnet_form_controller.js

```diff
@@ -61,27 +53,15 @@ ManageIQ.angular.app.controller('cloudSubnetFormController', ['$http', '$scope',
 
   $scope.filterNetworkManagerChanged = function(id) {
     miqService.sparkleOn();
-
-    $http.get('/cloud_subnet/cloud_subnet_networks_by_ems/' + id)
-      .then(getCloudSubnetNetworksByEmsComplete)
-      .catch(miqService.handleFailure);
-
-    $http.get('/cloud_subnet/cloud_tenants_by_ems/' + id)
-      .then(getCloudTenantsByEmsComplete)
-      .catch(miqService.handleFailure);
-
+    if (id) {
+      API.get("/api/cloud_networks?expand=resources&attributes=name,ems_ref&filter[]=ems_id=" + id).then(function(data) {
```

Comment:
@martinpovolny I'm not such a big fan of `CollectionsApi` because I think it is too naive.. not quite generic to be able to construct any URL, but not really understanding the API either.

But yeah, something like that could be useful..

maybe we can make it something like

```
API.get({
  collection: 'cloud_networks',
  expand_resources: true, // or expand: 'resources' but resources is magic, so I'd use expand only for an actual list of subcollections
  attributes: ['name', 'ems_ref'],
  filter: [ 'ems_id=' + id ],
})
```
but then we have to deal with syntax to be able to do `/collection`, `/collection/id` and `/collection/id/subcollection` etc...

but OTOH maybe all we need is a nice way of creating urls, possibly something like the upcoming  [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) standard..

```
var url = new URL('/api/cloud_networks');
url.searchParams.set('expand', 'resources');
url.searchParams.append('attributes', 'name');
url.searchParams.append('attributes', 'ems_ref');
url.searchParams.append('filter', 'ems_id=' + id);

API.get(url);
```

WDYT?